### PR TITLE
Dockerfile build args

### PR DESCRIPTION
**Pull request by Robert-Steiner**
**Summary**

- added the use build arguments to be able to specify the build profile (dev, release) and the feature set of the coordinator
- removed `docker.release` and `docker-compose-release.yml` file 
	- both became obsolete after the introduction of build arguments
- removed docker.release released stuff from ci 
- updated `.dockerignore` file 
	- added folders that do not have to be present in the Docker build context
- added the feature `full` as a shortcut for enabling all coordinator features
- updated README
- added `ENTRYPOINT` to dockerfile and removed `command` in `docker-compose` and `k8s` config

### CODE REVIEW

**Comment by Robert-Steiner** [comment]
#build

**Comment by Robert-Steiner** [comment]
#build

**Comment by Robert-Steiner** [comment]
#build

File: .github/workflows/dockercompose-validation.yml [hunk 1]
```diff
@@ -16,7 +16,3 @@ jobs:
       - name: Verify development docker-compose
```
Comment (rsaffi) [inline .github/workflows/dockercompose-validation.yml:16]: Given that there's only one `docker-compose.yml` now, then perhaps remove the reference to `development` from here as well?

File: .github/workflows/dockerfile-validation.yml [hunk 2]
```diff
@@ -15,6 +15,3 @@ jobs:
 
       - name: Lint dev file
```
Comment (rsaffi) [inline .github/workflows/dockerfile-validation.yml:16]: Same idea as previous comment.

**Review by rsaffi — APPROVED** [review]
I just put some minor comments regarding some stale references to "development" and "dev", but apart from that, this looks really nice! Great job! :rocket:

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -15,6 +15,9 @@ services:
       - xaynet
     ports:
       - "8081:8081"
+    # temporary fix:
+    # The coordinator crashes if the Redis is not ready or busy at startup
```
Comment (finiteprods) [inline docker/docker-compose.yml:19]:
```suggestion
    # The coordinator crashes if Redis is not ready or busy at startup
```

**Review by finiteprods — APPROVED** [review]

**Comment by Robert-Steiner** [comment]
Thanks @finiteprods and @rsaffi for reviewing 🙂

**Pull request merged**